### PR TITLE
Run CI against rails master

### PR DESCRIPTION
**Pull request by kmcphillips**
@thegedge @jonathankwok @sgrif 

https://twitter.com/sgrif/status/804037955241312258

Let's run our CI against rails master.

The matrix is silly big right now. But we can be opinionated in the 2.0 branch and drop support for anything below 4.2.

### CODE REVIEW

**Comment by sgrif** [comment]
Should master be added to allowed failures?

File: Gemfile.activesupport_master [hunk 1]
```diff
@@ -0,0 +1,4 @@
+source "https://rubygems.org"
+gemspec
+
+gem 'activesupport', github: 'rails/rails', branch: 'master'
```
Comment (sgrif) [inline Gemfile.activesupport_master]: No need to specify the branch

**Comment by sgrif** [comment]
Might want to look at using appraisal for this, makes a lot of it much easier.

**Comment by kmcphillips** [comment]
> Allowed failures

Probably not. This changes infrequently enough we should keep it working in all versions.

**Comment by kmcphillips** [comment]
https://github.com/thoughtbot/appraisal Interesting. I'll cut an issue to do that as part of the release milestone.

**Pull request merged**